internal/state/execution.go

```diff
@@ -347,22 +348,22 @@ func (blockExec *BlockExecutor) Commit(
 	err := blockExec.mempool.FlushAppConn(ctx)
 	if err != nil {
 		blockExec.logger.Error("client error during mempool.FlushAppConn", "err", err)
-		return nil, 0, err
+		return 0, err
 	}
 
 	// Commit block, get hash back
 	res, err := blockExec.appClient.Commit(ctx)
 	if err != nil {
 		blockExec.logger.Error("client error during proxyAppConn.Commit", "err", err)
-		return nil, 0, err
+		return 0, err
 	}
 
 	// ResponseCommit has no error code - just data
 	blockExec.logger.Info(
 		"committed state",
 		"height", block.Height,
 		"num_txs", len(block.Txs),
-		"app_hash", fmt.Sprintf("%X", res.Data),
+		"app_hash", appHash,
```

Comment:
We could probably remove that log here and just have it in `FinalizeBlock`

Reply:
Two things:
* I just realize I removed the formatting. That's probably bad. Will put it back
* To your question. I thought about it, and decided to pass the hash to this function _just_ to print it here, under the assumption that this is an important log people are used to seeing when troubleshooting. So didn't want to split it to minimize disruption. On the other hand, it might lead people to think the `app_hash` is still being returned at commit time.
So, I'm 50-50 on what to do... a little push from you in either direction and I will leave it or change it.

Reply:
OK, will log on `FinalizeBlock`